### PR TITLE
expat: patch CVE-2022-40674

### PR DESCRIPTION
**Pull request by BettyRain**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Fix the CVE-2022-40674

###### Change Log  

- Apply new patch for expat

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-40674

###### Test Methodology

- Local toolchain build succeeded, local full build succeeded
- Buddy build runs are successful: 
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=239907&view=results
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=239906&view=results

### CODE REVIEW

**Review by mfrw — CHANGES_REQUESTED** [review]
Can we please retarget to `main` instead of `2.0`

Also a better PR title could be:

`expat: patch CVE-2022-40674`

**Comment by BettyRain** [comment]
> Can we please retarget to `main` instead of `2.0`
> 
> Also a better PR title could be:
> 
> `expat: patch CVE-2022-40674`

Thank you! Fixed!

**Review by jslobodzian — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Review by christopherco — CHANGES_REQUESTED** [review]
Looks like expat is part of the raw toolchain build (toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh)

Do we need to patch the toolchain as well?

**Comment by BettyRain** [comment]
> Looks like expat is part of the raw toolchain build (toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh)
> 
> Do we need to patch the toolchain as well?

I think we don't patch this for 2.0

**Review by christopherco — APPROVED** [review]

File: SPECS/expat/expat.spec [hunk 1]
```diff
@@ -29,7 +30,7 @@ Group:          System Environment/Libraries
 This package contains minimal set of shared expat libraries.
 
 %prep
-%setup -q
+%autosetup -p2
```
Comment (mfrw) [inline SPECS/expat/expat.spec:33]: I am a little concerned about this `-p2` but LGTM otherwise

**Review by mfrw — APPROVED** [review]
LGTM 🥇 
Welcome to Mariner!

**Pull request merged**